torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -764,16 +764,15 @@ std::shared_ptr<at::cuda::CUDAEvent> ProcessGroupNCCL::CUDAEventCache::create(
   at::cuda::CUDAEvent* event = nullptr;
   {
     std::lock_guard<std::mutex> lock(cacheMutex_);
-    auto events = eventsArray_[timing ? 1 : 0];
+    auto& events = eventsArray_[timing ? 1 : 0];
```

Comment:
so there are 2 vectors of events, one for timing enabled and one for timing disabled

Reply:
yes. And each one is a std::vector.